src/dataset.jl

```diff
@@ -15,6 +15,16 @@ struct KeyedDataset{T<:XArray}
     constraints::OrderedSet{Pattern}
     # Data lookup can be by any type, but typically it'll either be symbol or tuple.
     data::LittleDict{Tuple{Vararg{Symbol}}, T}
+
```

Comment:
Could we add a docstring here to explain `check`?

Reply:
I'd rather not, as I see this is an implementation detail that shouldn't be part of the public API. I'm mostly just thinking of some performance oriented code that could be dangerous.